12-offer-encoding.md

```diff
@@ -0,0 +1,832 @@
+        * [`tu64`:`min`]
+    1. type: 22 (`offer_quantity_max`)
+    2. data:
+        * [`tu64`:`max`]
+    1. type: 24 (`offer_node_id`)
+    2. data:
+        * [`point`:`node_id`]
+
+1. subtype: `blinded_path`
+2. data:
+   * [`point`:`first_node_id`]
+   * [`point`:`blinding`]
+   * [`byte`:`num_hops`]
+   * [`num_hops*onionmsg_path`:`path`]
+
+## Requirements For Offers
+
+A writer of an offer:
+  - MUST NOT set any tlv fields greater or equal to 80, or tlv field 0.
+  - MUST set `offer_node_id` to the node's public key to request the invoice from.
+  - MUST set `offer_description` to a complete description of the purpose
+    of the payment.
+  - if the chain for the invoice is not solely bitcoin:
+    - MUST specify `offer_chains` the offer is valid for.
+  - otherwise:
+    - MAY omit `offer_chains`, implying that bitcoin is only chain.
+  - if a specific minimum `offer_amount` is required for successful payment:
+    - MUST set `offer_amount` to the amount expected (per item).
+    - if the currency for `offer_amount` is that of all entries in `chains`:
+      - MUST specify `amount` in multiples of the minimum lightning-payable unit
+        (e.g. milli-satoshis for bitcoin).
+    - otherwise:
+      - MUST specify `offer_currency` `iso4217` as an ISO 4712 three-letter code.
+      - MUST specify `offer_amount` in the currency unit adjusted by the ISO 4712
+        exponent (e.g. USD cents).
+  - MAY set `offer_metadata` for its own use.
+  - if it supports bolt12 offer features:
+    - MUST set `offer_features`.`features` to the bitmap of bolt12 features.
+  - if the offer expires:
+    - MUST set `offer_absolute_expiry` `seconds_from_epoch` to the number of seconds
+      after midnight 1 January 1970, UTC that invoice_request should not be
+      attempted.
+  - if it is connected only by private channels:
+    - MUST include `offer_paths` containing one or more paths to the node from
+      publicly reachable nodes.
+  - otherwise:
+    - MAY include `offer_paths`.
+  - if it includes `offer_paths`:
+    - SHOULD ignore any invoice_request which does not use the path.
+  - if it sets `offer_issuer`:
+    - SHOULD set it to identify the issuer of the invoice clearly.
+    - if it includes a domain name:
+      - SHOULD begin it with either user@domain or domain
+      - MAY follow with a space and more text
+  - if it can supply more than one item for a single invoice
+    - if the minimum quantity is more than 1:
+      - MUST set that minimum in `offer_quantity_min`
+    - if the maximum quantity is known:
+      - MUST set that maximum in `offer_quantity_max`
+    - if neither:
+      - MUST set `offer_quantity_min` to 1 to indicate `quantity` is supported.
+    - if both:
+      - MUST set `offer_quantity_min` less than or equal to `offer_quantity_max`.
+    - MUST NOT set `offer_quantity_min` or `offer_quantity_max` less than 1.
+
+A reader of an offer:
+  - if the offer contains any unknown TLV fields greater or equal to 80:
```

Comment:
Should it be "any" instead of  "any unknown" fields like when reading an `invoice_request`? The latter could be a bit ambiguous.

Reply:
Yes, and used that same sentence for the invoice_request requirements.